src/lib/conditionals/lightcone/5star/FlameOfBloodBlazeMyPath.ts

```diff
@@ -55,9 +49,22 @@ export default (s: SuperImpositionLevel, withContent: boolean): LightConeConditi
       const r = action.lightConeConditionals as Conditionals<typeof content>
 
       buffAbilityDmg(x, SKILL_DMG_TYPE | ULT_DMG_TYPE, (r.skillUltDmgBoost) ? sValuesSkillUltDmg[s] : 0, SOURCE_LC)
-      buffAbilityDmg(x, SKILL_DMG_TYPE | ULT_DMG_TYPE, (r.skillUltDmgBoost && r.bonusBoost) ? sValuesSkillUltDmg[s] : 0, SOURCE_LC)
     },
-    finalizeCalculations: () => {
+    finalizeCalculations: (x, action, context) => {
+      const r = action.lightConeConditionals as Conditionals<typeof content>
+      if (x.a[Key.HP] * sValuesHpDrain[s] > 500) {
+        buffAbilityDmg(x, SKILL_DMG_TYPE | ULT_DMG_TYPE, (r.skillUltDmgBoost) ? sValuesSkillUltDmg[s] : 0, SOURCE_LC)
+      }
+    },
+    gpuFinalizeCalculations: (action, context) => {
+      const r = action.lightConeConditionals as Conditionals<typeof content>
+      return `
+if (${wgslFalse(r.skillUltDmgBoost)}) {
+  return;
+}
```

Comment:
Tricky thing about wgsl injection, since finalizers get injected into the main loop you cant return from them otherwise it just kills the loop. So this has to be flipped into an if wgslTrue then do stuff